### ISSUE TITLE
Enable checking of all rules – usable for rule collections

### ISSUE THREAD

**Issue by voxpelli** [AriPerkkio/eslint-remote-tester#19]
I have a ESLint rule collection of my own, [`@voxpelli/eslint-config`](https://github.com/voxpelli/eslint-config), similar to eg. Standard (and extending that one) and I had hoped that I could use this module to achieve the same testing Standard does: https://github.com/standard/standard/blob/master/test/clone.js

That is: When I update a rule, or updates a plugin or an included rule collection, then I want to run this across lots of my repositories to see what is breaking and where that is breaking. This would make it much easier for me to update to newer plugin dependencies, as I can immediately find out any possible errors it might cause and decide on appropriate action.

So:

* I would want an option for a `rulesUnderTesting` that includes _everything_, pretty much a `testAllRules = true`

* A new `ignoreRulesWhenTesting` would be a good companion, to be able to filter out errors that are known to be correct.

Lastly so:

* It would be neat if the actual `.eslintrc` of the module itself was used, as sometimes they make some slight alterations to the base rules: https://github.com/voxpelli/node-connect-pg-simple/blob/main/.eslintrc

My target:

* Make it so that the CI of [automatic pull requests to my rule collection](https://github.com/voxpelli/eslint-config/pulls) can tell me from the start whether an updated dependency will affect my use cases or not

Would a PR for some of this be of interest?

**Comment by AriPerkkio** [comment]
Thanks for reaching out. Both of these feature requests sound like valid use cases and should be possible to implement.

### 1. Rule filtering:

The `rulesUnderTesting` could be extended so that it would accept a filtering function. This way it would cover the requirement of suggested `ignoreRulesWhenTesting` as well. 

```ts
type RulesFilter = (ruleId: string, options: { repository: string }) => boolean;

type Config = {
    ruleUnderTesting: string[] | RulesFilter;
    ...
}
```

https://github.com/AriPerkkio/eslint-remote-tester/blob/5cfd453fd44a0132b03ee8ad513dd8a917685698/lib/engine/worker-task.ts#L72-L75

This way all rules would be included with `{ rulesUnderTesting: () => true }` and a specific rule could be excluded by checking `ruleId` argument. The `options` argument is something that might be useful in future.

### 2. Using `eslintrc` from target repositories

This feature sounds intresting. Automatic detection of all variants of `eslintrc` is something I would like to avoid. Instead I would prefer to give users the ability to extend workflow with custom functionality. New configuration hook could cover this. 

```ts
type Config = {
    /** Optional callback invoked before creating linter for specific repository */
    onCreateLinter: (
        options: { 
            repository: string; // Repository under testing
            eslintrc: Linter.Config; // eslintrc from eslint-remote-tester.config.js
            path: string; // Path to the cloned repo
        }
    ) => Linter.Config | Promise<Linter.Config>;
}
```

The `options.path` is required since internal paths of `eslint-remote-tester` such as `.cache-eslint-remote-tester/{owner}/{repo}` are not part of public API. This option would ensure no breaking changes are introduced in case of internal changes.

This optional method would be called when creating Linter for the given repository using the output as `overrideConfig`.

https://github.com/AriPerkkio/eslint-remote-tester/blob/5cfd453fd44a0132b03ee8ad513dd8a917685698/lib/engine/worker-task.ts#L159-L162

This would allow users to modify `eslintrc` based on the current repository. For your case I could imagine something like below.

```js
// eslint-remote-tester.config.js
const fs = require('fs');

module.exports = {
  onCreateLinter: async function onCreateLinter(options) {
    if (options.repository === "some-speciic-repo") {
      const eslintrcFromRepo = fs.readFileSync(`${options.path}/.eslintrc.json`);

      return {
        ...options.eslintrc,
        rules: {
          ...options.eslintrc.rules,
          ...JSON.parse(eslintrcFromRepo).rules,
        },
      };
    }

    return optoions.eslintrc;
  },
};
```

I'm currently working on other features so PRs are welcome. 👍

**Comment by AriPerkkio** [comment]
@voxpelli this feature is available in `1.1.0`.

For your case I think configuration below would work:
```js
{
  eslintrc: { extends: "@voxpelli", root: true },
  rulesUnderTesting: function ruleFilter(ruleId, options) {
    if(ruleId === 'node/no-process-env' && options.repository === 'voxpelli/node-connect-pg-simple') {
      return false;
    }

    return true;
  }
}
```
